<launch>
    
    <!--- Sim Time -->
    <!-- <param name="/use_sim_time" default="false" /> -->
    <arg name="roll" default="0.0" />
    <arg name="pitch" default="0.0" />
    <arg name="yaw" default="0.0" />
    <arg name="x" default="0.0" />
    <arg name="y" default="0.0" />
    <arg name="z" default="0.0" />


    <!-- <arg name="roll" default="1.570795" />
    <arg name="pitch" default="0.0" />
    <arg name="yaw" default="1.570795" />
    <arg name="x" default="0.0" />
    <arg name="y" default="0.0" />
    <arg name="z" default="0.0" /> -->

    <!--- Run Rviz-->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ndt_localizer)/cfgs/rock-auto-gnss.rviz" />

    <!--- MapLoader -->    
<!--    <arg name="pcd_path"  default="$(find ndt_localizer)/map/0714opt/globalmap_lidar_feature.pcd"/>-->
    <arg name="pcd_path"  default="$(find ndt_localizer)/map/0925-1/globalmap_lidar_feature.pcd"/>
<!--    <arg name="pcd_path"  default="$(find ndt_localizer)/map/0925indoor/globalmap_lidar_feature.pcd"/>-->
<!--    <arg name="pcd_lla_path"  default="/home/lin/data/ndt-localizer-data/"/>-->
<!--    <arg name="pcd_lla_path"  default="$(find ndt_localizer)/map/0714opt/map_origin.txt"/>-->
    <arg name="pcd_lla_path"  default="$(find ndt_localizer)/map/0925-1/origin.txt"/>
<!--    <arg name="pcd_lla_path"  default="$(find ndt_localizer)/map/0925indoor/map_origin.txt"/>-->
<!--     <arg name="pcd_path"  default="$(find ndt_localizer)/map/ndt-localizer-data/kaist02.pcd"/>-->
    
    <arg name="map_topic" default="/ndt_localizer/points_map"/>
    <arg name="map_lla_topic" default="/ndt_localizer/map_lla"/>


    <node pkg="ndt_localizer" type="map_gnss_loader"    name="map_gnss_loader"    output="screen">
        <param name="pcd_path" value="$(arg pcd_path)"/>
        <param name="pcd_lla_path" value="$(arg pcd_lla_path)"/>
        <param name="map_topic" value="$(arg map_topic)"/>
        <param name="map_lla_topic" value="$(arg map_lla_topic)"/>

        <param name="roll" value="$(arg roll)" />
        <param name="pitch" value="$(arg pitch)" />
        <param name="yaw" value="$(arg yaw)" />
        <param name="x" value="$(arg x)" />
        <param name="y" value="$(arg y)" />
        <param name="z" value="$(arg z)" />
    </node>

</launch>
